src/roles/harlot.py

```diff
@@ -80,51 +77,48 @@ def on_transition_day_resolve(evt, var, victim):
 @event_listener("transition_day_resolve_end", priority=1)
 def on_transition_day_resolve_end(evt, var, victims):
     for victim in victims + evt.data["bitten"]:
-        if victim in evt.data["dead"] and victim.nick in VISITED.values() and (victim in evt.data["bywolves"] or victim in evt.data["bitten"]):
+        if victim in evt.data["dead"] and victim in VISITED.values() and (victim in evt.data["bywolves"] or victim in evt.data["bitten"]):
             for hlt in VISITED:
-                user = users._get(hlt) # FIXME
-                if VISITED[hlt] == victim.nick and user not in evt.data["bitten"] and user not in evt.data["dead"]:
+                if VISITED[hlt] is victim and hlt not in evt.data["bitten"] and hlt not in evt.data["dead"]:
                     if var.ROLE_REVEAL in ("on", "team"):
-                        evt.data["message"].append(messages["visited_victim"].format(hlt, get_reveal_role(hlt)))
+                        evt.data["message"].append(messages["visited_victim"].format(hlt, get_reveal_role(hlt.nick)))
                     else:
                         evt.data["message"].append(messages["visited_victim_noreveal"].format(hlt))
-                    evt.data["bywolves"].add(user)
-                    evt.data["onlybywolves"].add(user)
-                    evt.data["dead"].append(user)
+                    evt.data["bywolves"].add(hlt)
+                    evt.data["onlybywolves"].add(hlt)
+                    evt.data["dead"].append(hlt)
 
 @event_listener("transition_day_resolve_end", priority=3)
 def on_transition_day_resolve_end3(evt, var, victims):
     for harlot in get_all_players(("harlot",)):
-        if VISITED.get(harlot.nick) in list_players(var.WOLF_ROLES) and harlot not in evt.data["dead"] and harlot not in evt.data["bitten"]:
+        if VISITED.get(harlot) in get_players(var.WOLF_ROLES) and harlot not in evt.data["dead"] and harlot not in evt.data["bitten"]:
             evt.data["message"].append(messages["harlot_visited_wolf"].format(harlot))
             evt.data["bywolves"].add(harlot)
             evt.data["onlybywolves"].add(harlot)
             evt.data["dead"].append(harlot)
 
 @event_listener("night_acted")
 def on_night_acted(evt, var, user, actor):
-    if VISITED.get(user.nick):
+    if VISITED.get(user):
```

Comment:
can we rename these params from (evt, var, user, actor) to (evt, var, target, spy)? It is more descriptive as to what they actually represent -- spy is checking to see if target acted at night.